### DISCUSSION TITLE
Request for Alpaca market data driver

### DISCUSSION THREAD

**Discussion by kbmetz** [investbrainapp/investbrain#21] [Q&A] [answered]
Not sure if this should be a feature request, bug report or something else. But, it looks like the Yahoo market provider is already having issues. The curl is working but I'm getting a response back that includes 'Our engineers are working quickly to resolve the issue'. I can't tell if it's being throttled or if it's actually broken. Also there's nothing in the app to let me know there's an issue.

I've gone ahead and gotten an alphavantage API key to see if I can resolve the issue that way. But I see that limited to 25 requests a day.

There was a mention of other market providers. I'd like to put in a request for https://alpaca.markets since I have an account with them and I know they do API's.

**Comment by hackeresq** [comment, marked as answer]
Nice, hadn't heard of Alpaca. Adding to the roadmap #15 now!

Yahoo has sliced both ways - free, easy, no config required. But it has been cracking down. The [downstream package](https://github.com/scheb/yahoo-finance-api) Investbrain relies on for Yahoo has been playing cat and mouse for months.

**Comment by kbmetz** [comment]
How long should it take before I see a graph? I added in the alphavantage API key and did a docker compose down/up but still not seeing any updates anywhere on the pages. Also tried setting it to 1M and still don't see any changes or updates as far as I can tell.

**Reply by hackeresq** [reply]
The daily capture command is run daily. But you can force fill the graph by running:

```bash
docker exec -it investbrain-app php artisan sync:daily-change
```

**Reply by MRobi1** [reply]
Would it make sense to have this command ran automatically in the background with the workflow for adding a new transaction? This way the graph updates automatically for users when adding a transaction?

**Reply by hackeresq** [reply]
Yes, great call! This happens automatically now. Whenever a transaction that would affect the market price, the daily change table is updated automatically!

**Comment by hackeresq** [comment]
Alpaca was added in v1.2.0
